Assets/Scripts/Gamestate/AIManager.cs

```diff
@@ -181,12 +252,24 @@ void Update()
 #endif
         animator.SetFloat("Forward", Vector3.Dot(agent.velocity, transform.forward) / agent.speed);
         animator.SetFloat("Right", Vector3.Dot(agent.velocity, transform.right) / agent.speed);
+        if (agent.isOnOffMeshLink)
+            onLinkStart?.Invoke();
         if (!ShootingTarget)
             return;
         GunOrigin.LookAt(ShootingTarget.position, transform.up);
         Fire();
     }
 
+    private void FixedUpdate()
+    {
+        if (isDead || !agent.enabled)
+            return;
+        var nextPosition = agent.nextPosition;
+        transform.position = Vector3.Lerp(transform.position, nextPosition, agent.speed * Time.fixedDeltaTime);
+        if (updateRotation)
+            transform.LookAt(new Vector3(ShootingTarget ? ShootingTarget.position.x : nextPosition.x, transform.position.y, ShootingTarget ? ShootingTarget.position.z : nextPosition.z), transform.up);
```

Comment:
Perhaps leave the ternary outside the vector constructor

Reply:
Will do!